File: features/home/account-home/src/main/java/com/tokopedia/home/account/di/component/BuyerAccountComponent.java
package com.tokopedia.home.account.di.component;

import com.tokopedia.abstraction.common.di.component.BaseAppComponent;
import com.tokopedia.home.account.di.module.BuyerAccountModule;
import com.tokopedia.home.account.di.scope.BuyerAccountScope;
import com.tokopedia.home.account.presentation.fragment.BuyerAccountFragment;

import dagger.Component;

/**
 * @author okasurya on 7/17/18.
 */
@Component(modules = {BuyerAccountModule.class}, dependencies = {BaseAppComponent.class})
@BuyerAccountScope
public interface BuyerAccountComponent {
    void inject(BuyerAccountFragment fragment);
}

File: features/home/account-home/src/main/java/com/tokopedia/home/account/di/component/BuyerAccountComponent.kt
package com.tokopedia.home.account.di.component

import com.tokopedia.abstraction.common.di.component.BaseAppComponent
import com.tokopedia.home.account.di.module.BuyerAccountModule
import com.tokopedia.home.account.di.module.BuyerAccountViewModelModule
import com.tokopedia.home.account.di.module.RevampedBuyerAccountModule
import com.tokopedia.home.account.di.scope.BuyerAccountScope
import com.tokopedia.home.account.presentation.fragment.BuyerAccountFragment
import com.tokopedia.recommendation_widget_common.di.RecommendationModule

import dagger.Component

/**
 * @author okasurya on 7/17/18.
 */
@Component(modules = [BuyerAccountModule::class,
    BuyerAccountViewModelModule::class,
    RecommendationModule::class,
    RevampedBuyerAccountModule::class], dependencies = [BaseAppComponent::class])
@BuyerAccountScope
interface BuyerAccountComponent {
    fun inject(fragment: BuyerAccountFragment)
}
